src/Gaffer/GraphComponent.cpp

```diff
@@ -471,6 +473,92 @@ const GraphComponent::ChildContainer &GraphComponent::children() const
 	return m_children;
 }
 
+void GraphComponent::reorderChildren( const ChildContainer &newOrder )
+{
+	if( newOrder.size() != m_children.size() )
+	{
+		throw IECore::InvalidArgumentException(
+			boost::str( boost::format( "Wrong number of children specified (%1% but should be %2%)" ) % newOrder.size() % m_children.size() )
+		);
+	}
+
+	// Build map from child to index, so we can quickly look up the
+	// index for a particular child.
+
+	unordered_map<const GraphComponent *, size_t> indexMap;
+	size_t index = 0;
+	for( const auto &c : m_children )
+	{
+		indexMap[c.get()] = index++;
+	}
```

Comment:
Addressed in 2205124. I could go either way on this, but limiting the scope of the `index` variable clinched it for me. I do find it a bit awkward either way, and it is something where Python's `enumerate()` is a nice solution. But trying to emulate `enumerate()` in C++ just leads to more complexity. I briefly considered this, which I may even have used somewhere else once, but I guess it might be a bit obfuscated?

```
for( const auto &c : m_children )
{
     indexMap[c.get()] = &c - &m_children.front();
}
```